pkg/catalog/manager/crd.go

```diff
@@ -24,6 +23,13 @@ const (
 	TemplateNameLabel = "catalog.cattle.io/template_name"
 )
 
+type CatalogManagerType struct {
+	catalog        *v3.Catalog
+	projectCatalog *v3.ProjectCatalog
+	clusterCatalog *v3.ClusterCatalog
+	catalogType    string
```

Comment:
why do we need a type here? assuming that the presence of one of the pointers would be enough to figure this information out, catalogTypes seems to be redundant, and you have to remember to set it always